R/dataProcessPlots.R

```diff
@@ -491,4 +602,140 @@ dataProcessPlots = function(
     }
     .saveTable(result, address, "ConditionPlot_value")
   }
+  if (isPlotly) {
+      return(plots)
+  }
+}
+
+#' converter for plots from ggplot to plotly
+#' @noRd
+.convert.ggplot.plotly = function(plot, tips = "all") {
+    converted_plot <- ggplotly(plot,tooltip = tips)
+    converted_plot <- converted_plot %>% 
+        plotly::layout(
+            width = 800,   # Set the width of the chart in pixels
+            height = 600,  # Set the height of the chart in pixels
+            title = list(
+                font = list(
+                    size = 18
+                )
+            ),
+            xaxis = list(
+                titlefont = list(
+                    size = 15  # Set the font size for the x-axis label
+                )
+            ),
+            legend = list(
+                x = 0,     # Set the x position of the legend
+                y = -0.25,    # Set the y position of the legend (negative value to move below the plot)
+                orientation = "h",  # Horizontal orientation
+                font = list(
+                    size = 12  # Set the font size for legend item labels
+                ),
+                title = list(
+                    font = list(
+                        size = 12  # Set the font size for the legend title
+                    )
+                )
+            )
+        ) 
+    return(converted_plot)
+}
+
+.fix.legend.plotly.plots.dataprocess = function(plot) {
+    df <- data.frame(id = seq_along(plot$x$data), legend_entries = unlist(lapply(plot$x$data, `[[`, "name")))
+    df$legend_group <- gsub("^\\((.*?),.*", "\\1", df$legend_entries)
+    df$is_first <- !duplicated(df$legend_group)
+    df$is_bool <- ifelse(grepl("TRUE|FALSE", df$legend_group), TRUE, FALSE)
+    # df[nrow(df), "is_first"] <- FALSE 
+    plot$x$data[[nrow(df)]]$showlegend <- FALSE # remove text legend
+    for (i in df$id) {
+        is_first <- df$is_first[[i]]
+        is_bool <- df$is_bool[[i]]
+        plot$x$data[[i]]$name <- df$legend_group[[i]]
+        plot$x$data[[i]]$legendgroup <- plot$x$data[[i]]$name
+        if (!is_first) plot$x$data[[i]]$showlegend <- FALSE
+        if(is_bool) plot$x$data[[i]]$showlegend <- FALSE
+    }
+    plot
+
+}
+
+.fix.censored.points.legend.profileplots.plotly = function(plot) {
+    df <- data.frame(id = seq_along(plot$x$data), legend_entries = unlist(lapply(plot$x$data, `[[`, "name")))
+    detected_data = FALSE
+    censored_data = FALSE
+    for (i in df$id) {
+        if(!detected_data && df$legend_entries[[i]] == "FALSE") {
+            plot$x$data[[i]]$name <- "Detected data"
+            plot$x$data[[i]]$showlegend <- TRUE
+            # plot$x$data[[i]]$marker$color <- 'rgba(0,0,0,1)'
```

Comment:
will this code be used in the final version?

Reply:
Commit [better handling of cencored data code](https://github.com/Vitek-Lab/MSstats/pull/107/commits/e3ebec22a4dd9bf8806ee21c21bdef90a5b93af5)

This code is better and readable, basically, we are altering plotly code to combine multiple data points into 2 unique points, censored missing data and detected data